src/update-done/update-done.c

```diff
@@ -31,8 +31,8 @@ static int apply_timestamp(const char *path, struct timespec *ts) {
 
         r = write_string_file_atomic_label_ts(path, message, ts);
         if (r == -EROFS)
-                return log_debug_errno(r, "Cannot create \"%s\", file system is read-only.", path);
-        if (r < 0)
+                log_debug_errno(r, "Cannot create \"%s\", file system is read-only.", path);
```

Comment:
yeah, this looks right, and is also what the original version did in 8ea48dfcd33e8db0c01bf8c57c3bbcfdc3c86d4b